Source/EasyNetQ/Interception/InterceptorRegistrator.cs

```diff
@@ -2,32 +2,39 @@
 
 namespace EasyNetQ.Interception
 {
+    /// <summary>
+    ///     Helper interface that allows to register interceptors in DI
+    /// </summary>
     public interface IInterceptorRegistrator
     {
+        /// <summary>
+        /// Registers an instance of <see cref="IProduceConsumeInterceptor"/> interceptor
+        /// </summary>
         void Add(IProduceConsumeInterceptor interceptor);
+
+        /// <summary>
+        /// Registers an interceptor of the specified type <typeparamref name="TInterceptor"/>
+        /// </summary>
+        void Add<TInterceptor>() where TInterceptor : class, IProduceConsumeInterceptor;
     }
 
-    public class InterceptorRegistrator : IInterceptorRegistrator
+    internal sealed class InterceptorRegistrator : IInterceptorRegistrator
     {
-        private readonly CompositeInterceptor compositeInterceptor;
         private readonly IServiceRegister serviceRegister;
 
         public InterceptorRegistrator(IServiceRegister serviceRegister)
         {
             this.serviceRegister = serviceRegister;
-            compositeInterceptor = new CompositeInterceptor();
         }
 
-        public IServiceRegister Register()
+        public void Add(IProduceConsumeInterceptor interceptor)
         {
-            serviceRegister.Register<IProduceConsumeInterceptor>(compositeInterceptor);
-            return serviceRegister;
+            serviceRegister.Register(interceptor);
```

Comment:
@Pliner  note that this line will be changed to call `RegisterMore` after #1108 .